skl2onnx/operator_converters/TextVectorizer.py

```diff
@@ -46,16 +120,45 @@ def convert_sklearn_text_vectorizer(scope, operator, container):
             "only stip_accents=None is supported.")
 
     options = container.get_options(
-            op, dict(sep=[' ', '.', '?', ',', ';', ':', '!']))
-    if set(options) != {'sep'}:
+            op, dict(sep="DEFAULT",
+                     regex=None))
+    if set(options) != {'sep', 'regex'}:
         raise RuntimeError("Unknown option {} for {}".format(
                                 set(options) - {'sep'}, type(op)))
-    default_pattern = '(?u)\\b\\w\\w+\\b'
-    default_separators = options['sep']
-    if op.token_pattern != default_pattern:
-        raise NotImplementedError(
-            "Only the default tokenizer based on default regular expression "
-            "'{0}' is implemented.".format(default_pattern))
+
+    if op.analyzer == 'word':
+        default_pattern = '(?u)\\b\\w\\w+\\b'
+        if options['sep'] == "DEFAULT" and options['regex'] is None:
+            warnings.warn("Converter for TfidfVectorizer will use "
+                          "scikit-learn regular expression by default "
+                          "in version 1.6.",
+                          DeprecationWarning)
+            default_separators = [' ', '.', '?', ',', ';', ':', '!']
+            regex = op.token_pattern
+            if regex == default_pattern:
+                regex = '[a-zA-Z0-9_]+'
+            default_separators = None
+        elif options['regex'] is not None:
+            if options['regex']:
+                regex = options['regex']
+            else:
+                regex = op.token_pattern
+                if regex == default_pattern:
+                    regex = '[a-zA-Z0-9_]+'
+            default_separators = None
+        else:
+            regex = None
+            default_separators = options['sep']
+    else:
+        if options['sep'] != 'DEFAULT':
+            raise RuntimeError("Option sep has not effect "
+                               "if analyser != 'word'.")
+        if options['regex']:
```

Comment:
There 4 lines could be shortened to:
regex = options['regex'] if options['regex'] else '.'
(more Pythonic)

Reply:
Fixed but I'd like flake8 to capture this otherwise let's leave it that way.